### PR TITLE
2024-01-23 release notes

### PR DESCRIPTION
**Pull request by shashankbarsin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,59 @@
...
+* Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks:~:text=exception%20YAML.-,Install%20Azure%20Policy%20Add%2Don%20for%20AKS,-Before%20you%20install) will now no longer support the validation for constraint template. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/).
```
Comment (wangyira) [inline CHANGELOG.md:9]: Could this link be updated[the Azure Policy Add-On](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+
+* Preview features
+  * Istio revision 1.19 is now available with Istio-based service mesh add-on. More information on performing canary upgrade for the new minor revision of Istio can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade). Default revision of the Istio service mesh add-on for new clusters has been updated to 1.18. Istio 1.17 version is no longer supported.
+  * Istio based service mesh addon now supports plugin CA to allow users to provide their own certificates and keys for signing workload certificates. More information can be found [here](https://aka.ms/asm-plugin-ca)
+  * When troubleshooting AKS nodes, for developers not having access to Kubernetes API but having access to node ARM API, node IP and name information are now made available in this API. More information on accessing the nodes using the private IPs can be found [here](https://learn.microsoft.com/azure/aks/node-access#create-an-interactive-shell-connection-to-a-node-using-the-ip-address).
+
+* Bug Fixes 
+  * When Standard_HB120-16rs_v3 SKU is selected, `singleplacementgroup: true` is not set for VMSS in the node pool.
```
Comment (wangyira) [inline CHANGELOG.md:24]: what's the intended behavior? Signleplacementgroup should be set for true, was it set to false before or just empty

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,60 @@
...
+ 
+### Release notes
+
+* Features
+  * Ability to update an existing [Kubenet based AKS cluster](https://learn.microsoft.com/azure/aks/configure-kubenet) to use [Azure CNI Overlay](https://learn.microsoft.com/azure/aks/azure-cni-overlay) is now generally available. More information can be found [here](https://learn.microsoft.com/azure/aks/azure-cni-overlay?tabs=kubectl#upgrade-an-existing-cluster-to-cni-overlay)
+  * The application routing add-on can now manage multiple public and internal NGINX ingress controller. [Advanced ingress controller configuration](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration) is possible via a Custom Resource Definition (CRD).
+
+* Preview features
+  * Istio revision 1.19 is now available with Istio-based service mesh add-on. More information on performing canary upgrade for the new minor revision of Istio can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade). Default revision of the Istio service mesh add-on for new clusters has been updated to 1.18. Istio 1.17 version is no longer supported.
```
Comment (kaarthis) [inline CHANGELOG.md:20]: Should we document what happens if someone where to use 1.17 - any error link?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,60 @@
...
+  * Istio based service mesh addon now supports plugin CA to allow users to provide their own certificates and keys for signing workload certificates. More information can be found [here](https://aka.ms/asm-plugin-ca)
+  * When troubleshooting AKS nodes, for developers not having access to Kubernetes API but having access to node ARM API, node IP and name information are now made available in this API. More information on accessing the nodes using the private IPs can be found [here](https://learn.microsoft.com/azure/aks/node-access#create-an-interactive-shell-connection-to-a-node-using-the-ip-address).
```
Comment (kaarthis) [inline CHANGELOG.md:22]: node ip and node name

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,60 @@
...
+
+* Bug Fixes 
+  * When Standard_HB120-16rs_v3 SKU is selected, `singleplacementgroup: true` is not set for VMSS in the node pool.
+  * Fixed nodeAffinity in calico-node DaemonSet to prevent scheduling on virtual kubelet nodes
```
Comment (kaarthis) [inline CHANGELOG.md:26]: GH link?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,60 @@
...
+
+* Behavioral Change
+  * [Network observability addon](https://learn.microsoft.com/azure/aks/network-observability-overview) charts are updated with following changes:
+    * Introduced CPU (500m) and Memory (300Mi) limits for `kappie container` in `kappie-agent` DaemonSet
+    * Introduced a new `init-kappie` init container as part of `kappie-agent` DaemonSet
+    * api-resources `nodes` and `namespaces` added to `kappie-cluster-reader` ClusterRole
+  * [Vertical pod autoscaling](https://learn.microsoft.com/azure/aks/vertical-pod-autoscaler) charts are updated with following changes:
+    * A new containerPort on 8943 named `prometheus`  added to `overlay-vpa-cert-webhook-check` job's `overlay-vpa-webhook-generation` container to expose prometheus metrics
+    * Liveness and readiness probes added for addon components
+  * Metrics port 29615 added for Azure File CSI driver on Linux node on AKS 1.28
```
Comment (kaarthis) [inline CHANGELOG.md:37]: please add a period at the end of every bullet.

**Review by kaarthis — COMMENTED** [review]
PTAL

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,60 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* All current AKS API versions silently ignore unknown fields. An unknown field is a field that isn't part of the AKS API. AKS API version 2024-01-01, 2024-01-02-preview and all subsequent API versions will change this behavior. Unknown fields in a request will result in the request being rejected with an error stating that the unknown field is not understood. This change only impacts new API versions and won't impact you unless you update to use an API version 2024-01-01 or later. Existing API calls (via Azure Resource Manager templates or otherwise) will continue to function as-is.
```
[the remark this answers is not in the document]
Comment (shashankbarsin) [reply CHANGELOG.md:11]: above text is the same as what we send in comms for subscriptions where unknown fields. the actual unknown fields that were detected are sent over mail to the subscription owners

File: CHANGELOG.md (hunk 3 above)
Comment (shashankbarsin) [reply CHANGELOG.md:20]: It's the same as AKS. 1.17 is no longer available for new installations. For existing users of 1.17, we don't force anything and it's already documented in the above linked doc that all minor revision upgrades are to be initiated by user. So if the user stays on 1.17 past its EOL and faces an issue with Istio, then we'd not be able to provide support

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,60 @@
...
+  * Added `appgw.ingress.azure.io` api-group to ` ingress-appgw-cr` ClusterRole to address missing api-group permissions error in [Application Gateway Ingress Controller addon](https://learn.microsoft.com/azure/application-gateway/ingress-controller-overview) container.
+
+* Behavioral Change
+  * [Network observability addon](https://learn.microsoft.com/azure/aks/network-observability-overview) charts are updated with following changes:
+    * Introduced CPU (500m) and Memory (300Mi) limits for `kappie container` in `kappie-agent` DaemonSet
+    * Introduced a new `init-kappie` init container as part of `kappie-agent` DaemonSet
+    * api-resources `nodes` and `namespaces` added to `kappie-cluster-reader` ClusterRole
+  * [Vertical pod autoscaling](https://learn.microsoft.com/azure/aks/vertical-pod-autoscaler) charts are updated with following changes:
+    * A new containerPort on 8943 named `prometheus`  added to `overlay-vpa-cert-webhook-check` job's `overlay-vpa-webhook-generation` container to expose prometheus metrics
+    * Liveness and readiness probes added for addon components
```
[the remark this answers is not in the document]
Comment (shashankbarsin) [reply CHANGELOG.md:36]: this bullet point is nested under VPA addon, so only applicable to that. no link as the probes are not meant to be modified by users, rather they are added only to ensure better health probes for the addon components

**Review by qpetraroia — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,60 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-01-23
+
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).
+
+### Announcements
+* Kubernetes 1.25 is being deprecated on January 14, 2024 and support will transition to our [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy). Please upgrade to Kubernetes version 1.26 or above.
```
Comment (palma21) [inline CHANGELOG.md:8]: Passed?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,60 @@
...
+* Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy addon](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks) will now no longer support the validation for constraint template. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/).
```
Comment (palma21) [inline CHANGELOG.md:9]: Already landed/ behavior change?

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,60 @@
...
+  * Istio based service mesh addon now supports plugin CA to allow users to provide their own certificates and keys for signing workload certificates. More information can be found [here](https://aka.ms/asm-plugin-ca).
+  * When troubleshooting AKS nodes, for developers not having access to Kubernetes API but having access to node ARM API, node IP and node name information are now made available in this API. More information on accessing the nodes using the private IPs can be found [here](https://learn.microsoft.com/azure/aks/node-access#create-an-interactive-shell-connection-to-a-node-using-the-ip-address).
+  * The application routing add-on can now manage multiple public and internal NGINX ingress controllers. [Advanced ingress controller configuration](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration) is possible via a Custom Resource Definition (CRD).
+
+* Bug Fixes 
+  * When Standard_HB120-16rs_v3 SKU is selected, `singleplacementgroup: true` is now set for VMSS in the node pool.
```
Comment (palma21) [inline CHANGELOG.md:25]: this property is not visible for users. We need this translated to what the visible issue/behavior was
(which is lack of IB support)

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,60 @@
...
+  * Fixed nodeAffinity in calico-node DaemonSet to prevent scheduling on virtual kubelet nodes.
+  * Added `appgw.ingress.azure.io` api-group to ` ingress-appgw-cr` ClusterRole to address missing api-group permissions error in [Application Gateway Ingress Controller addon](https://learn.microsoft.com/azure/application-gateway/ingress-controller-overview) container.
+
+* Behavioral Change
+  * [Network observability addon](https://learn.microsoft.com/azure/aks/network-observability-overview) charts are updated with following changes:
+    * Introduced CPU (500m) and Memory (300Mi) limits for `kappie container` in `kappie-agent` DaemonSet.
+    * Introduced a new `init-kappie` init container as part of `kappie-agent` DaemonSet.
+    * api-resources `nodes` and `namespaces` added to `kappie-cluster-reader` ClusterRole.
+  * [Vertical pod autoscaling](https://learn.microsoft.com/azure/aks/vertical-pod-autoscaler) charts are updated with following changes:
```
Comment (palma21) [inline CHANGELOG.md:34]: Addon, not charts

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,60 @@
...
+    * A new containerPort on 8943 named `prometheus`  added to `overlay-vpa-cert-webhook-check` job's `overlay-vpa-webhook-generation` container to expose prometheus metrics.
```
Comment (palma21) [inline CHANGELOG.md:35]: Better to simplify/explain what this means and make this a feature. 

You can now scrape prom metrics from the VPA addon for example

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,60 @@
...
+    * Liveness and readiness probes added for addon components.
```
Comment (palma21) [inline CHANGELOG.md:36]: should not be relevant to the user

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,60 @@
...
+  * Metrics port 29615 added for Azure File CSI driver on Linux node on AKS 1.28.
```
Comment (palma21) [inline CHANGELOG.md:37]: User facing behavior? (should be similar to above?)

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,60 @@
...
+  * Fixed networking observability agent crashing issue on Windows node pool of AKS clusters version >= 1.28.
```
Comment (palma21) [inline CHANGELOG.md:38]: Should this be bundled in the first point of NW obs?

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,60 @@
...
+  * Upgraded Azure Monitor Metrics to v.6.8.3. Full changelog can be found [here](https://github.com/Azure/prometheus-collector/blob/main/RELEASENOTES.md#release-01-09-2024).
+  * Upgraded ip-masq-agent-v2 to [v0.1.9](https://github.com/Azure/ip-masq-agent-v2/releases/tag/v0.1.9).
+  * Upgraded aad-pod-identity/nmi image from v1.8.17-1 to v1.8.18 to address golang.org/x/net CVEs - [CVE-2023-39325](https://nvd.nist.gov/vuln/detail/CVE-2023-39325), [CVE-2023-3978](https://nvd.nist.gov/vuln/detail/CVE-2023-3978), and [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487)
+  * Upgraded Azure workload identity image to [v1.2.0](https://github.com/Azure/azure-workload-identity/releases/tag/v1.2.0).
+  * Upgraded vertical pod autoscaler recommender image to [v0.12.0](https://github.com/kubernetes/autoscaler/releases/tag/vertical-pod-autoscaler-0.12.0).
+  * AKS extension in VS Code has been updated to [1.4.1](https://github.com/Azure/vscode-aks-tools/releases/tag/1.4.1).
```
Comment (palma21) [inline CHANGELOG.md:52]: Not a cluster component, suggest moving to feature or preview feature

**Review by palma21 — CHANGES_REQUESTED** [review]

**Review by kaarthis — APPROVED** [review]

File: CHANGELOG.md (hunk 12 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 13 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 14 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: removed

File: CHANGELOG.md (hunk 16 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 17 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: moved

File: CHANGELOG.md (hunk 11 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: updated to mention Infiniband

File: CHANGELOG.md (hunk 10 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: moved under behavior change

**Review by palma21 — APPROVED** [review]

**Comment by mojtaba3d** [comment]
the latest Image which is available for AKS in WE is "202401.09.0". I don't get any "202401.17.0". any idea ?

**Pull request merged**